src/IO/H5/CombineH5.cpp

```diff
@@ -149,4 +157,186 @@ void combine_h5_vol(const std::vector<std::string>& file_names,
     new_file.close_current_object();
   }
 }
+
+void combine_h5_dat(const std::vector<std::string>& h5_files_to_combine,
```

Comment:
I think you assume that the files are in increasing order in time. It would be good to add a check for that since otherwise users may be very surprised. Another option would be to just sort the list in increasing order for the subfile(s).